app/controllers/api/encryptables_controller.rb

```diff
@@ -121,4 +124,12 @@ def permitted_attrs
       permitted_attrs
     end
   end
+
+  def log_read_access
+    encryptable.paper_trail.save_with_version
+    v = encryptable.versions.first
```

Comment:
relying on first here does not sound very safe, does it?

this also creates a complete Version entry with the whole object every time an encryptable is being red.

can't we create a manual entry here for read access? something like:
```
PaperTrail::Version.create!(item_id: encryptable_id, item_type. .... event: :viewed, whodunnit ....)
```

Reply:
I've touched upon this above already (https://github.com/puzzle/cryptopus/pull/587#discussion_r862608971), it just hasn't been implemented yet.
`encryptable.paper_trail.save_with_version` already returns the PaperTrail::Version thingy so we can just do this:
`v = encryptable.paper_trail.save_with_version`